DQMOffline/Trigger/plugins/RazorMonitor.cc

```diff
@@ -0,0 +1,371 @@
+  genericTriggerEventPSet.add<bool>("andOrDcs", false);
+  genericTriggerEventPSet.add<bool>("errorReplyDcs", true);
+  genericTriggerEventPSet.add<std::string>("dbLabel","");
+  genericTriggerEventPSet.add<bool>("andOrHlt", true);
+  genericTriggerEventPSet.add<edm::InputTag>("hltInputTag", edm::InputTag("TriggerResults::HLT") );
+  genericTriggerEventPSet.add<std::vector<std::string> >("hltPaths",{});
+  genericTriggerEventPSet.add<std::string>("hltDBKey","");
+  genericTriggerEventPSet.add<bool>("errorReplyHlt",false);
+  genericTriggerEventPSet.add<unsigned int>("verbosityLevel",1);
+
+  desc.add<edm::ParameterSetDescription>("numGenericTriggerEventPSet", genericTriggerEventPSet);
+  desc.add<edm::ParameterSetDescription>("denGenericTriggerEventPSet", genericTriggerEventPSet);
+
+  //binning from 2016 offline selection
+  edm::ParameterSetDescription histoPSet;
+  std::vector<double> mrbins = {0., 100., 200., 300., 400., 500., 575., 650., 750., 900., 1200., 1600., 2500., 4000.};
+  histoPSet.add<std::vector<double> >("mrBins", mrbins);
+
+  std::vector<double> rsqbins = {0., 0.05, 0.1, 0.15, 0.2, 0.25, 0.30, 0.41, 0.52, 0.64, 0.8, 1.5};
+  histoPSet.add<std::vector<double> >("rsqBins", rsqbins);
+
+  std::vector<double> dphirbins = {0., 0.5, 1.0, 1.5, 2.0, 2.5, 2.8, 3.0, 3.2};
+  histoPSet.add<std::vector<double> >("dphiRBins", dphirbins);
+
+  edm::ParameterSetDescription lsPSet;
+  fillHistoLSPSetDescription(lsPSet);
+  histoPSet.add<edm::ParameterSetDescription>("lsPSet", lsPSet);
+
+  desc.add<edm::ParameterSetDescription>("histoPSet",histoPSet);
+
+  descriptions.add("razorMonitoring", desc);
+}
+
+//CalcMR and CalcR borrowed from HLTRFilter.cc
+double RazorMonitor::CalcMR(const math::XYZTLorentzVector& ja, const math::XYZTLorentzVector& jb){
+  if(ja.Pt()<=0.1) return -1;
+
+  double A = ja.P();
+  double B = jb.P();
+  double az = ja.Pz();
+  double bz = jb.Pz();
+  TVector3 jaT, jbT;
+  jaT.SetXYZ(ja.Px(),ja.Py(),0.0);
+  jbT.SetXYZ(jb.Px(),jb.Py(),0.0);
+  double ATBT = (jaT+jbT).Mag2();
+
+  double MR = sqrt((A+B)*(A+B)-(az+bz)*(az+bz)-
+                   (jbT.Dot(jbT)-jaT.Dot(jaT))*(jbT.Dot(jbT)-jaT.Dot(jaT))/(jaT+jbT).Mag2());
+  double mybeta = (jbT.Dot(jbT)-jaT.Dot(jaT))/
+    sqrt(ATBT*((A+B)*(A+B)-(az+bz)*(az+bz)));
+
+  double mygamma = 1./sqrt(1.-mybeta*mybeta);
+
+  //use gamma times MRstar
+  return MR*mygamma;
+}
+
+double RazorMonitor::CalcR(double MR, const math::XYZTLorentzVector& ja, const math::XYZTLorentzVector& jb, const edm::Handle<std::vector<reco::PFMET> >& inputMet, const std::vector<math::XYZTLorentzVector>& muons){
+  //now we can calculate MTR
+  TVector3 met;
+  met.SetPtEtaPhi((inputMet->front()).pt(),0.0,(inputMet->front()).phi());
+
+  std::vector<math::XYZTLorentzVector>::const_iterator muonIt;
+  for (muonIt = muons.begin(); muonIt!=muons.end(); muonIt++) {
+    TVector3 tmp;
+    tmp.SetPtEtaPhi(muonIt->pt(), 0, muonIt->phi());
+    met-=tmp;
+  }
+
+  TVector3 ja3, jb3;
```

Comment:
In trying to remove these TVector3's, I find that XYZTLorentzVector doesn't have a member called momentum(), and if I use Vect(), I get a ROOT::Math::DisplacementVector3D which has a + operator but no Dot().
Probably I am missing something; could you please clarify? Thanks.